app/alignment/utils.py

```diff
@@ -413,3 +415,93 @@ def log_solution_choice(solution_choice_uuid, shared_solutions):
         raise DatabaseError(
             message="An error occurred while saving user b's solution choice to the db."
         )
+
+
+def build_shared_impact_details_response(impact_iri):
+    """
+    Get the detals for a shared impact.
+
+    Includes checks that the IRI is found and of the correct type, e.g. not a solution IRI.
+
+    Parameters
+    ==========
+    impact_iri - the IRI for the shared impact
+
+    Returns
+    ==========
+    JSON:
+    - the effect title
+    - the image url (if available)
+    - the long description for the effect
+    - the sources for the effect
+    - a list of personal values that are related to the effect
+    """
+
+    G = current_app.config["G"].copy()
+    nx = network_x_utils()
+    impact = None
+
+    for node in G.nodes:
+        if G.nodes[node]["iri"][24:] == impact_iri:
+            nx.set_current_node(G.nodes[node])
+            impact = {
+                "effectTitle": G.nodes[node]["label"],
+                "imageUrl": nx.get_image_url_or_none(),
+                "longDescription": nx.get_description(),
+                "effectSources": nx.get_causal_sources(),
+                "relatedPersonalValues": map_associated_personal_values(
+                    G.nodes[node]["personal_values_10"]
+                ),
+            }
+            break
+
+    if not impact:
+        raise NotInDatabaseError(message="The shared impact IRI cannot be found.")
+    elif len(impact["effectSources"]) == 0:
```

Comment:
Sometimes an impact might not have a source for some reason. Checking to make sure the title exists might be a better way to check that the impact exists?

Reply:
I meant to ask you if it could be assumed that there are sources. Obviously it can't. I added this check with the thought that if someone mixed up impact and solution IRIs while testing, they would get that message because the function to pull sources for those different types is different, while other functions like for pulling the title are shared.

But note, the first check will see if the IRI exists at all. The second check is only type validation of sorts.